From: Antonio Alvarez Feijoo <antonio.feijoo@suse.com>
Date: Wed, 8 Jul 2026 08:53:02 +0200
Subject: fix(crypt): do not call return to exit a systemd generator

systemd generators implemented via shell scripts are not sourced by systemd, so
calling `return` within them to exit is incorrect.

Fixes #2490
Origin: upstream, https://github.com/dracut-ng/dracut/commit/fbb7270edcf9fbc1c60959dc35e532dfe5c382b8
---
 modules.d/70crypt/crypt-generator.sh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/modules.d/70crypt/crypt-generator.sh b/modules.d/70crypt/crypt-generator.sh
index 5c65f4c..f543e6a 100755
--- a/modules.d/70crypt/crypt-generator.sh
+++ b/modules.d/70crypt/crypt-generator.sh
@@ -22,13 +22,13 @@ generator_set_device_timeout() {
 
 if ! getargbool 1 rd.luks; then
     # crypto LUKS detection is disabled
-    return 0
+    exit 0
 fi
 
-[ -e /etc/crypttab ] || return 0
+[ -e /etc/crypttab ] || exit 0
 
 GENERATOR_DIR="$1"
-[ -n "$GENERATOR_DIR" ] || return 1
+[ -n "$GENERATOR_DIR" ] || exit 1
 [ -d "$GENERATOR_DIR" ] || mkdir -p "$GENERATOR_DIR"
 
 timeout=$(getarg rd.timeout)
